tests/test_datatypes.py

```diff
@@ -30,3 +30,22 @@
 def test_datatype_pickling(dtype) -> None:
     copy_dtype = copy.deepcopy(dtype)
     assert copy_dtype == dtype
+
+
+@pytest.mark.parametrize(
+    ["source", "expected"],
+    [
+        (str, DataType.string()),
```

Comment:
perhaps could test 
- `dict` (our `DataType::Map`)
- `tuple` (our `DataType::Struct` I think with just arbitrary names)
- `bytearray`

Reply:
trust your judgement here though :)

Reply:
Oh yes the dict map one is a good idea. Let me add that.